rosidl_generator_c/CMakeLists.txt

```diff
@@ -13,33 +13,10 @@ endif()
 
 find_package(ament_cmake_python REQUIRED)
 find_package(ament_cmake_ros REQUIRED)
-find_package(rosidl_typesupport_interface REQUIRED)
-
-include_directories(include)
-add_library(${PROJECT_NAME}
-  "src/message_bounds.c"
-  "src/message_type_support.c"
-  "src/primitives_sequence_functions.c"
-  "src/service_type_support.c"
-  "src/string_functions.c"
-  "src/u16string_functions.c"
-)
-ament_target_dependencies(${PROJECT_NAME}
-  "rosidl_typesupport_interface")
-if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
-  set_target_properties(${PROJECT_NAME} PROPERTIES
-    COMPILE_OPTIONS -Wall -Wextra -Wpedantic)
-endif()
-if(WIN32)
-  target_compile_definitions(${PROJECT_NAME}
-    PRIVATE "ROSIDL_GENERATOR_C_BUILDING_DLL")
-endif()
+find_package(rosidl_runtime_c REQUIRED)
 
 ament_export_dependencies(rosidl_cmake)
 ament_export_dependencies(rosidl_typesupport_interface)
```

Comment:
Is this still necessary - since the dependency has been removed from the manifest?

Reply:
tests need to link against this dependency

Reply:
Then the dependency can't be removed from the manifest.

Reply:
Also this line only exports the package to downstream packages. It doesn't make it available to the tests in this repo.

Reply:
Restored

Reply:
The commit 83e43928555bdb8300bf29dde5b5ac25cb8eeea0 is incorrect. `rosidl_typesupport_interface` is not being used at build time of the package but only exported to downstream packages. So the `build_depend` should be removed but the `build_export_depend` must be restored.